src/components/ElementBar/ColorPickerButton/extractFirstNonTransparentOrFirstColor.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Elements } from '../../../state/types';
+
+/**
+ * @return Looks for the first element in the list without a transparent colour. If found it returns this color.
+ *         Otherwise it falls back to the colour of the first element, that may also be "transparent".
+ *         Returns undefined if the colour cannot be determined.
+ */
+export const extractFirstNonTransparentOrFirstColor = (
+  elements: Elements,
+): string | undefined => {
```

Comment:
Please provide jsdoc for all parameters.
Should be simply a function.

Reply:
Docs for parameters added. Also refactored the function to use early returns.

Reply:
> Should be simply a function.

It is currently unclear to me what that means. Isn't it a function?

Reply:
please use `function` keyword here, I meant to define function using `function` keyword

Reply:
Ah okay. :+1:  I think this is commonly known as a „regular function“ compared to „arrow function“.